sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -98,7 +99,11 @@ class UnivocityParser(
     legacyFormat = FAST_DATE_FORMAT,
     isParsing = true)
 
-  private val csvFilters = new CSVFilters(filters, requiredSchema)
+  private val csvFilters = if (SQLConf.get.csvFilterPushDown) {
```

Comment:
Shall we match it in JSON side too?

Reply:
Let's do that in a separate PR since JSON filters are not related to Avro.

Reply:
Here is the PR https://github.com/apache/spark/pull/29206